index.js

```diff
@@ -134,11 +134,40 @@ var Router = function(){
 
       var route = Route(path);
       route.fn = fn;
+      route.idx = this.routes.length;
```

Comment:
This may cause bad situations in case of

* Adding 4 Routes will give us idx's of [0,1,2,3]
* Remove Route with idx=1 leaves us with [0,2,3]
* Adding a new Route gives us [0,2,3,3]

It's better to have a private incrementor, something based off `Date.now()` or something based off `Math.random()`